pipeline/scripts/15-dIEM_violin.R

```diff
@@ -386,6 +397,22 @@ if (violin == 1) { # make violin plots
     metab_interest_controls <- prepare_data(metab_list_all, zscore_controls)
     metab_perpage <- prepare_data_perpage(metab_interest_sorted, metab_interest_controls, nr_plots_perpage, nr_pat, nr_contr)
     
+    # for Diagnostics metabolites to be saved in Helix
+    if(grepl("Diagnost", pdf_dir)) {
+      # get table that combines DIMS results with stofgroepen/Helix table
+      DIMS_Helix_table <- get_patient_data_to_Helix(metab_interest_sorted, metab_list_all)
+      
+      # check if run contains Diagnostics patients (e.g. "P2024M"), not for research runs
+      if(any(grepl("^P[0-9]{4}M", DIMS_Helix_table$Patient))){
```

Comment:
This regex is used in `create_violin_plots.R` as well -> would be wise to create a function `is_diagnostics_patient`.

Reply:
Agree, added function